core/unit_test/hip/TestHIP_Category.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 2.0
+//              Copyright (2014) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_TEST_HIP_HPP
+#define KOKKOS_TEST_HIP_HPP
+
+#include <gtest/gtest.h>
+
+namespace Test {
+
+class hip : public ::testing::Test {
+protected:
+  static void SetUpTestCase() {
+  }
+
+  static void TearDownTestCase() {
+  }
+};
+
+} // namespace Test
```

Comment:
Let's bring it up at developers meeting and suggest we rip these out everywhere in Kokkos.

Reply:
I am not sure what you mean here. Should it look like this:
```c++
#define TEST_CATEGORY ::testing::Test

class hip : public TEST_CATEGORY
{
protected:
  static void SetUpTestCase() {}
  static void TearDownTestCase(){}
};
```
I assume these functions are pure virtual.

Reply:
I don't get what you want to do here. I tried changing the code like you wanted but I get a ton of errors.

Reply:
I meant 
```C++
#ifndef KOKKOS_TEST_HIP_HPP
#define KOKKOS_TEST_HIP_HPP

#include <gtest/gtest.h>

#define TEST_CATEGORY ::testing::Test
#define TEST_EXECSPACE Kokkos::Experimental::HIP

#endif
```
I thought it would work when looking [here](https://github.com/google/googletest/blob/84d986531e8df9fb6ea4a7bae8fb130c05b3782c/googletest/include/gtest/gtest.h#L427-L428) and [here](https://github.com/google/googletest/blob/84d986531e8df9fb6ea4a7bae8fb130c05b3782c/googletest/include/gtest/gtest.h#L2348-L2350) but I haven't tried

Reply:
I tried that but I got a ton of errors